include/circt/Dialect/HW/HWOps.h

```diff
@@ -104,11 +104,19 @@ bool isAnyModule(Operation *module);
 /// Return the signature for the specified module as a function type.
 FunctionType getModuleType(Operation *module);
 
-/// Return the number of inputs for the specified module.
-unsigned getModuleNumInputs(Operation *module);
+/// Return the number of inputs for the specified module/instance.
+inline unsigned getModuleNumInputs(Operation *moduleOrInstance) {
+  assert(isAnyModule(moduleOrInstance) &&
```

Comment:
This should work on instances as well.  Please add a new `isAnyModuleOrInstance` predicate that admits InstanceOp.  We can then use that to simplify the assert at the top of `getAllModulePortInfos`'s body and use it here.